### ISSUE TITLE
Panic when passing empty `Vec<Series>` to `Series::new`

### ISSUE THREAD

**Issue by mt-caret** [pola-rs/polars#10561]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Reproducible example

```rust
let vec: Vec<Series> = Vec::new();

Series::new("column", vec)
```
(https://github.com/mt-caret/polars-ocaml/blob/551c993cf538e5749c7022f8c3f486c945949916/rust/polars-ocaml/src/lib.rs#L186)

### Issue description

The issue stems from the fact that `Series::new` assumes the vector has at least one element in it, which isn't guaranteed: 
https://github.com/pola-rs/polars/blob/b1361fe487c4544f26a7f32db60d65c46d92a5a4/crates/polars-core/src/named_from.rs#L126

Unfortunately due to how the API works it seems difficult to actually return a series when there are no elements in the vec since we can't figure out the type; does that mean it's impossible to create empy list series?

### Expected behavior

Create an empty series, similar to when a vec of any over values are passed to `Series::new`.

### Installed versions

```toml
features = [
    "approx_unique",
    "asof_join",
    "describe",
    "diagonal_concat",
    "dot_diagram",
    "dtype-date",
    "dtype-datetime",
    "dtype-i16",
    "dtype-i8",
    "dtype-u16",
    "dtype-u8",
    "dynamic_groupby",
    "horizontal_concat",
    "interpolate",
    "json",
    "lazy",
    "list_eval",
    "parquet",
    "pivot",
    "random",
    "rank",
    "round_series",
    "semi_anti_join",
    "streaming",
    "timezones",
]
```

**Comment by c-peters** [comment]
related to #10458

**Comment by c-peters** [comment]
We could give back List[DataType::Null] similar to https://github.com/pola-rs/polars/blob/b1361fe487c4544f26a7f32db60d65c46d92a5a4/crates/polars-core/src/named_from.rs#L147-L163

**Comment by orlp** [comment]
Also related to https://github.com/pola-rs/polars/pull/10558.

**Comment by coastalwhite** [comment]
This is also closed from #10558.
